src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Editor/Listeners/FrameOpenCloseListener.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Editor
+{
+    [Export(typeof(IFrameOpenCloseListener))]
+    internal class FrameOpenCloseListener : OnceInitializedOnceDisposedAsync, IFrameOpenCloseListener, IVsWindowFrameEvents, IVsSolutionEvents
+    {
+        private readonly IServiceProvider _serviceProvider;
+        private readonly IEditorStateModel _editorModel;
+        private readonly IProjectThreadingService _threadingService;
+        private readonly UnconfiguredProject _unconfiguredProject;
+        private IVsWindowFrame _frame;
+        private uint _eventCookie = VSConstants.VSCOOKIE_NIL;
+        private uint _solutionEventCookie = VSConstants.VSCOOKIE_NIL;
+
+        [ImportingConstructor]
+        public FrameOpenCloseListener(
+            [Import(typeof(SVsServiceProvider))]IServiceProvider helper,
+            IEditorStateModel editorModel,
+            IProjectThreadingService threadingService,
+            UnconfiguredProject unconfiguredProject) :
+#pragma warning disable IDE0030 // Use null propagation https://github.com/dotnet/roslyn/issues/16114
+            base(threadingService != null ? threadingService.JoinableTaskContext : throw new ArgumentNullException(nameof(threadingService)))
+#pragma warning restore IDE0030 // Use null propagation
+        {
+            Requires.NotNull(helper, nameof(helper));
+            Requires.NotNull(editorModel, nameof(editorModel));
+            Requires.NotNull(unconfiguredProject, nameof(unconfiguredProject));
+
+            _serviceProvider = helper;
+            _editorModel = editorModel;
+            _threadingService = threadingService;
+            _unconfiguredProject = unconfiguredProject;
+        }
+
+        public Task InitializeEventsAsync(IVsWindowFrame frame)
+        {
+            Requires.NotNull(frame, nameof(frame));
+            _frame = frame;
+            return InitializeAsync();
+        }
+
+        public void OnFrameDestroyed(IVsWindowFrame frame)
+        {
+            if (_frame.Equals(frame))
+            {
+                _threadingService.ExecuteSynchronously(_editorModel.DisposeEditorAsync);
```

Comment:
Who observes this? I think we throw this on at least a silent watson if we don't expect this to throw.